checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -1188,34 +1285,104 @@ private boolean isMustCallClose(Node node) {
     return typeFactory.getMustCallValues(mcAtf.withoutClose(mustCallAnnotation)).isEmpty();
   }
 
+  /**
+   * Add a new alias to all Obligations that have {@code var} in their resource-alias set. This
+   * method should be used when {@code var} and {@code newAlias} definitively point to the same
+   * object in memory.
+   *
+   * @param obligations the set of Obligations to modify
+   * @param var a variable
+   * @param newAlias a new {@link ResourceAlias} to add
+   */
+  private void addAliasToObligationsContainingVar(
+      Set<Obligation> obligations, LocalVariableNode var, ResourceAlias newAlias) {
+    Iterator<Obligation> it = obligations.iterator();
+    List<Obligation> newObligations = new ArrayList<>();
+
+    while (it.hasNext()) {
+      Obligation obligation = it.next();
+      if (obligation.canBeSatisfiedThrough(var)) {
+        it.remove();
+
+        Set<ResourceAlias> newAliases = new LinkedHashSet<>(obligation.resourceAliases);
+        newAliases.add(newAlias);
+
+        newObligations.add(new Obligation(newAliases, obligation.whenToEnforce));
+      }
+    }
+
+    obligations.addAll(newObligations);
+  }
+
   /**
    * Remove any Obligations that contain {@code var} in their resource-alias set.
    *
-   * @param obligations the set of Obligations
+   * @param obligations the set of Obligations to modify
    * @param var a variable
    */
   private void removeObligationsContainingVar(Set<Obligation> obligations, LocalVariableNode var) {
-    Obligation obligationForVar = getObligationForVar(obligations, var);
-    while (obligationForVar != null) {
-      obligations.remove(obligationForVar);
-      obligationForVar = getObligationForVar(obligations, var);
-    }
+    removeObligationsContainingVar(
+        obligations, var, MustCallAliasHandling.NO_SPECIAL_HANDLING, MethodExitKind.ALL);
+  }
+
+  /**
+   * Helper type for {@link #removeObligationsContainingVar(Set, LocalVariableNode,
+   * MustCallAliasHandling, Set)}
+   */
+  private enum MustCallAliasHandling {
```

Comment:
Is there a reason we need an `enum` here and this can't just be a boolean parameter to `removeObligationsContainingVar()`?

Reply:
No reason other than [personal stylistic preference](https://apaloski.github.io/coding/style/enum-vs-boolean).

At a glance, (I think) this code is more difficult to read:
```java
          removeObligationsContainingVar(
              ...
              false, // no shred of a clue what this argument does :(
              ...);
```

Reply:
Interesting, a two-valued enum to kind of simulate a named parameter!  I'm fine with keeping it here 🙂